ui/components/app/nft-details/nft-full-image.js

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useHistory, useParams } from 'react-router-dom';
+import { getAssetImageURL } from '../../../helpers/utils/util';
+import { getNftImageAlt } from '../../../helpers/utils/nfts';
+import { getCurrentNetwork, getIpfsGateway } from '../../../selectors';
+
+import {
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  IconName,
+} from '../../component-library';
+import { NftItem } from '../../multichain/nft-item';
+import { Content, Header, Page } from '../../multichain/pages/page';
+
+import { getNfts } from '../../../ducks/metamask/metamask';
+import { isEqualCaseInsensitive } from '../../../../shared/modules/string-utils';
+import {
+  Display,
+  IconColor,
+  JustifyContent,
+} from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { ASSET_ROUTE } from '../../../helpers/constants/routes';
+
+export default function NftFullImage() {
+  const t = useI18nContext();
+  const { asset, id } = useParams();
+  const nfts = useSelector(getNfts);
+  const nft = nfts.find(
+    ({ address, tokenId }) =>
+      isEqualCaseInsensitive(address, asset) && id === tokenId.toString(),
+  );
+
+  const { image, imageOriginal, name, tokenId } = nft;
+
+  const ipfsGateway = useSelector(getIpfsGateway);
+  const currentChain = useSelector(getCurrentNetwork);
+
+  const nftImageAlt = getNftImageAlt(nft);
+  const nftSrcUrl = imageOriginal ?? image;
+  const nftImageURL = getAssetImageURL(imageOriginal ?? image, ipfsGateway);
+  const isIpfsURL = nftSrcUrl?.startsWith('ipfs:');
+  const isImageHosted = image?.startsWith('https:');
+  const history = useHistory();
+
+  const [visible, setVisible] = useState(false);
+
+  useEffect(() => {
+    setVisible(true);
+  }, []);
```

Comment:
```suggestion
  const [visible, setVisible] = useState(false);

  useEffect(() => {
    setVisible(true);
  }, []);
```
can be replaced with
```
const [visible, setVisible] = useState(true);
```

Reply:
Don't think this suggestion would work;
I need the visible to be initially false, and then after the component is rendered set it to true to have the transition effect.